lib/ansible/module_utils/redfish_utils.py

```diff
@@ -1904,6 +1904,138 @@ def get_multi_nic_inventory(self, resource_type):
+
+        response = self.get_request(self.root_uri + resource_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+
+        if key not in data:
+            return {'ret': False, 'msg': "Key %s not found" % key}
+
+        serial_interfaces_uri = data[key]["@odata.id"]
+
+        # Get a list of all serial interfaces and build respective URIs
+        response = self.get_request(self.root_uri + serial_interfaces_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+
+        for serial_interface in data[u'Members']:
+            serial_interface_list.append(serial_interface[u'@odata.id'])
+
+        for n in serial_interface_list:
+            serial_interface = {}
+            uri = self.root_uri + n
+            response = self.get_request(uri)
+            if response['ret'] is False:
+                return response
+            data = response['data']
+
+            for property in properties:
+                if property in data:
+                    serial_interface[property] = data[property]
+
+            serial_interface_results.append(serial_interface)
+
+        result["entries"] = serial_interface_results
+        return result
+
+    def get_multi_serialinterfaces(self):
+        ret = True
+        entries = []
+
+        resource_uris = self.manager_uris
+
+        for resource_uri in resource_uris:
+            serialinterfaces = self.get_serial_interfaces(resource_uri)
+            ret = serialinterfaces.pop('ret') and ret
+            if 'entries' in serialinterfaces:
+                entries.append(({'resource_uri': resource_uri},
+                               serialinterfaces['entries']))
+        return dict(ret=ret, entries=entries)
+
+    def set_serial_interface(self, serial_interface_id, serial_interface_config):
+        # Get SerialInterfaces collection
+        response = self.get_request(self.root_uri + self.manager_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+        if 'SerialInterfaces' not in data:
+            return {'ret': False, 'msg': "SerialInterfaces resource not found"}
+        serial_interfaces_uri = data["SerialInterfaces"]["@odata.id"]
+        response = self.get_request(self.root_uri + serial_interfaces_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+        serial_interface_uris = []
+        for serial_interface in data[u'Members']:
+            serial_interface_uris.append(serial_interface[u'@odata.id'])
+
+        # Find target Serial Interface
+        target_serial_interface_uri = None
+        target_serial_interface_current_setting = None
+        for uri in serial_interface_uris:
+            # If serial_interface_id is not specified, set first interface
+            if serial_interface_id in ['null', '']:
```

Comment:
For this case (no id specified; use the first one), it looks like there is some code missing to read the data from the first URI and then set `target_serial_interface_current_setting` to the data.

Reply:
Many thanks, Bill, good catch!  I have modified the code, please help to review again, thanks!